neal/sampler.py

```diff
@@ -235,13 +235,10 @@ def interrupt_function():
         num_betas = int(math.ceil(sweeps / sweeps_per_beta))
         if beta_schedule_type == "linear":
             # interpolate a linear beta schedule
-            beta_step = (beta_range[1] - beta_range[0]) / float(num_betas)
-            beta_schedule = [beta_range[0] + s * beta_step
-                             for s in range(num_betas)]
+            beta_schedule = np.linspace(*beta_range, num=num_betas)
         elif beta_schedule_type == "geometric":
-            beta_step = (beta_range[1] / beta_range[0]) ** (1.0 / num_betas)
-            beta_schedule = [beta_range[0] * beta_step ** i
-                             for i in range(num_betas)]
+            # interpolate a geometric beta schedule
+            beta_schedule = np.logspace(*np.log(beta_range), num=num_betas)
```

Comment:
Correct me if I'm wrong, but I believe you're simply replacing the old 3 lines of code (old242-244) with a single new line 241? If that's the case, I believe the single new line should actually be:
```
schedule_steps = np.logspace(0, num_betas-1, base=(beta_range[1]/beta_range[0])**(1/num_betas), num=num_betas)
beta_schedule = beta_range[0] * schedule_steps
```
Although, I'm more of a fan of an INCLUSIVE beta range, which means changing my second arg from `num_betas-1` to `num_betas`

Reply:
Actually. the CLEANER and SHORTER code you probably want (and if you don't mind an inclusive range) is:
```
np.geomspace(*beta_range, num=num_betas)
```
Otherwise, if we want to produce the same geometric schedule as before (non-inclusive range), then:
```
np.geomspace(*beta_range, num=num_betas, endpoint=False)
```

Reply:
Cool! I didn't know about `geomspace`. Let's go with that!

I always forget that default base of `logspace` is 10 but `log` is e. Don't know why they did that. What I really meant was `np.logspace(*np.log10(beta_range), num=num_betas)`.